pkg/api/index.go

```diff
@@ -230,9 +230,20 @@ func (hs *HTTPServer) setIndexViewData(c *m.ReqContext) (*dtos.IndexViewData, er
 				}
 
 				if include.Type == "page" && include.AddToNav {
-					link := &dtos.NavLink{
-						Url:  setting.AppSubUrl + "/plugins/" + plugin.Id + "/page/" + include.Slug,
-						Text: include.Name,
+					var link *dtos.NavLink
+					if len(include.Path) > 0 {
+						link = &dtos.NavLink{
+							Url:  setting.AppSubUrl + include.Path,
+							Text: include.Name,
+						}
```

Comment:
I am not sure I understand this, what is path here? an app linking to any grafana page? a plugin page needs to follow a specific route pattern `/plugins/<plugin_id>/page/<pagename>`  not sure how this change fixes react pages?

Reply:
Since react "root" pages have full control over:
```
/a/<plugin_id>/*
```

And the config pages have control over:
```
/plugins/<plugin_id>/?page=*&<whatever params>
```

This approach reuses the existing 'path' variable and lets you include a link to any path in grafana.  

The assumption is that the plugin author knows the path and parameters it wants to load